### ISSUE TITLE
uiautomator:BasicSample test fail 

### ISSUE THREAD

**Issue by longkai** [googlesamples/android-testing#18]
I clone the code and run the test with command ``./gradlew clean connectedAndroidTest`` in root project. The uiautomator:BasicSample always fail.

below is the fail message from gradle.

```
com.example.android.testing.uiautomator.BasicSample.ChangeTextBehaviorTest > checkPreconditions[Nexus 5 - 5.1] FAILED 
	java.lang.IllegalStateException: UiAutomationService android.accessibilityservice.IAccessibilityServiceClient$Stub$Proxy@3ce0391calready registered!
	at android.os.Parcel.readException(Parcel.java:1554)

com.example.android.testing.uiautomator.BasicSample.ChangeTextBehaviorTest > testChangeText_newActivity[Nexus 5 - 5.1] FAILED 
	java.lang.IllegalStateException: UiAutomation not connected!
	at android.app.UiAutomation.throwIfNotConnectedLocked(UiAutomation.java:914)

com.example.android.testing.uiautomator.BasicSample.ChangeTextBehaviorTest > testChangeText_sameActivity[Nexus 5 - 5.1] FAILED 
	java.lang.IllegalStateException: UiAutomation not connected!
	at android.app.UiAutomation.throwIfNotConnectedLocked(UiAutomation.java:914)
Tests on Nexus 5 - 5.1 failed: Instrumentation run failed due to 'java.lang.IllegalStateException'
:uiautomator:BasicSample:app:connectedAndroidTest FAILED

FAILURE: Build failed with an exception.
```

**Comment by otw** [comment]
Sounds like UiAutomator is not able to connect to UiAutomation. I can't reproduce it locally. I will investigate a bit more. Does rebooting the device help to solve the issue?

**Comment by longkai** [comment]
After I rebooted my Nexus 5 and ran the test again, strange thing happened.

```
com.example.android.testing.espresso.BasicSample.ChangeTextBehaviorTest > changeText_newActivity[Nexus 5 - 5.1] FAILED 
        android.support.test.espresso.base.DefaultFailureHandler$AssertionFailedWithCauseError: 'with text: is "Espresso"' doesn't match the selected view.
        Expected: with text: is "Espresso"

com.example.android.testing.espresso.BasicSample.ChangeTextBehaviorTest > changeText_sameActivity[Nexus 5 - 5.1] FAILED 
        android.support.test.espresso.base.DefaultFailureHandler$AssertionFailedWithCauseError: 'with text: is "Espresso"' doesn't match the selected view.
        Expected: with text: is "Espresso"
:espresso:BasicSample:app:connectedAndroidTest FAILED

FAILURE: Build failed with an exception.
```

This test passed yesterday.

**Comment by longkai** [comment]
I figured out why the last test fail because the device' s default keyboard language is non-English at that time. I switched to English and the whole project also passed.

However, the keyboard is English yesterday and back to the first issue, I still don' t konw why after the rebooting all the tests passed.

In addition, can we choose the right language keyborad when we want to type something in the test?

Thx.
